Source/controls/plrctrls.cpp

```diff
@@ -257,9 +257,9 @@ void FindMeleeTarget()
 		SearchNode node = queue.front();
 		queue.pop_front();
 
-		for (int i = 0; i < 8; i++) {
-			const int dx = node.x + pathxdir[i];
-			const int dy = node.y + pathydir[i];
+		for (auto pathDir : PathDirs) {
+			const int dx = node.x + pathDir.deltaX;
+			const int dy = node.y + pathDir.deltaY;

```

Comment:
```suggestion
			const auto testPosition = node + pathDir;
			const int dx = testPosition.x;
			const int dy = testPosition.y;
```
To handle changing the backing type of PathDirs

Reply:
PathNode could also be changed to be composed of a Point to allow for this directly.